src/GraphQL/Types/ResolveFieldContextAs.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using System.Threading;
+using GraphQL.Instrumentation;
+using GraphQL.Language.AST;
+
+namespace GraphQL.Types
+{
+    internal class ResolveFieldContextAs<T> : IResolveFieldContext<T>
```

Comment:
I think that `ReadonlyResolveFieldContext`, `ResolveFieldContextAs` and `ResolveFieldContext` all should be in one namespace.

Reply:
Also consider to rename `ResolveFieldContextAs` to `ResolveFieldContextAdapter`.

Reply:
Sure I can move it, but realize that the ReadonlyResolveFieldContext is designed exclusively for use with the ExecutionNode and ExecutionContext classes.  The other three are in the GraphQL.Types namespace, so I'll move it there.

Reply:
`ResolveFieldContextAdapter` does sound better

Reply:
Done - moved ReadonlyResolveFieldContext, and renamed ResolveFieldContextAs to ResolveFieldContextAdapter